src/item/enchantment/EnchantmentOption.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item\enchantment;
+
+class EnchantmentOption{
+
+	/**
+	 * @param EnchantmentInstance[] $enchantments
+	 */
+	public function __construct(
+		private int $networkId,
```

Comment:
This seems redundant, as it's informed by the position in the `options` array anyway

It would make more sense to pass the XP level cost here, which would also allow custom costs

Reply:
Yes, we can remove $networkId from EnchantmentOption. Then in `NetworkSession::sendEnchantmentOptions` optionId will be set from 0 to 2 for options. Although I don't know, maybe in some situations the networkId would be useful, otherwise why would it be needed in a packet. 

But enchantment level is exactly the XP level cost. And it is the enchantment level that is determined by the position in the `options` in the array. So there's no way to set a custom XP level cost.